test-image

```diff
@@ -14,19 +15,20 @@ docker build -t rust-musl-builder-using-diesel examples/using-diesel
 docker run --rm rust-musl-builder-using-diesel
 
 # Make sure we can build a static executable.
-docker run --rm ekidd/rust-musl-builder bash -c "
-set -euo pipefail
-export USER=rust
-cargo new --vcs none --bin testme
-cd testme
-cargo build
+rm -rf test-project/target/
+docker run -v "$(pwd)"/test-project:/home/rust/src --rm ekidd/rust-musl-builder cargo build --release
```

Comment:
One drawback of using `-v` like this for tests is that it creates a whole bunch of files (typically owned by a different user ID that may not even exist on the host system) which are outside the container and hard to clean up afterwards.

Is there any reason why the original "run a script inside the container" code wasn't working? I can think of several alternatives to `-v`:

1. Put back the original code.
2. Turn the original code into a shell script which gets built into the test image, and then just call it as a one-liner.
3. Use `docker cp` to extract the file you want to look at rather than using `-v`.

Reply:
It is not about extracting the file. I want to mount the test project. It is a lot simpler to mount one than to create it inside the container.

Reply:
It also works on travis. They use user 2000 and it is not a problem.

Reply:
I really don't want to use `-v`. I've run into way too many problems related to `-v` and various CI systems over the years, and I've decided that (for my projects at least) that CI systems should just avoid mounting directories into containers. Yes, I've gotten `-v` to work many, many times, but I always end up having to revisit it 6 to 12 months later. On average, I get bitten by `-v`-related surprises or behavior changes about twice a year.

Please consider one of the other 3 options I proposed.

Reply:
I'm sorry I don't want to spend more time on this. If it bothers you that much please just cherry-pick the bug fix and ignore the rest.
I don't really see the problem though.

Reply:
What's the worry with `-v`? If you run this on travis the only thing that might happen is that the binary ends up with weird permissions, but that's a cleanup issue. Since travis literally throws away the VM / namespace after building, I don't see how this could go wrong.

I mean the alternative is encoding the project in terms of echo etc, or using docker cp which has just so many more ways to blow up because the container has to exist for cp to work. I'd also have expected "drop the test project folder in the sources and mount it" to be the simplest and most robust solution.

One could do the ldd check in the container I guess, would that be good enough for you?

Reply:
I'm not interested in using `-v` in any CI-related project that I maintain, unless there's no good way to avoid it. This year alone, I've been bitten twice by `-v` mounts on various projects:

1. The change from uid 1000 to 2000 on TravisCI broke a number of build systems for me.
2. A recent version of Docker changed how permissions worked on certain `-v` mounts, once again breaking a bunch of other stuff that I maintained.

And this is not the first time I've had problems with `-v`. So as a matter of personal convenience, I've been removing `-v` from all the CI systems I maintain. I do not want to reintroduce it here.

The pre-existing version of this code (not including the PR) works perfectly fine for me.

Please feel free to choose one of the 3 alternatives above.